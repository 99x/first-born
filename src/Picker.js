import React, { Component } from "react";
import { StyleSheet, Platform, Picker, View } from "react-native";
import createReactClass from "create-react-class";
import Ionicon from "react-native-vector-icons/Ionicons";

class PickerAE extends Component {

    constructor(props) {
        super(props);
        this.state = {
            focused: false
        }
    }

    render() {
        const { color, ...otherProps } = this.props;

        const underLineColor = !color ? "#000" : color;

        if (Platform.OS === "android") {
            return (
                <View
                    style={this.state.focused ? [styles.pickerAndroid, { borderBottomColor: underLineColor, borderBottomWidth: 2 }] : [styles.pickerAndroid, { borderBottomColor: underLineColor, borderBottomWidth: 1 }]}
                    onResponderGrant={() => this.setState({ focused: true })}
                    onResponderTerminate={() => this.setState({ focused: false })}
                >
                    <Picker
                        ref={c => (this._root = c)}
                        mode="dropdown"
                        {...otherProps}>
                        {otherProps.children}
                    </Picker>
                    <Ionicon style={styles.icon} size={24} color={underLineColor} name={this.state.focused ? "md-arrow-dropup" : "md-arrow-dropdown"} />
                </View>
            )
        }
        return (
            <View style={styles.pickerIos}>
                <Picker
                    ref={c => (this._root = c)}
                    {...otherProps}>
                    {otherProps.children}
                </Picker>
            </View>
        )
    }
}

PickerAE.Item = createReactClass({
    render() {
        return <Picker.Item {...this.props} />;
    }
});

PickerAE.propTypes = {
    ...Picker.propTypes
}

const styles = StyleSheet.create({
    pickerAndroid: {
        width: "100%",
        borderTopRightRadius: 3,
        borderTopLeftRadius: 3,
        marginVertical: 10,
    },
    pickerIos: {
        backgroundColor: "white",
        borderRadius: 10,
        paddingHorizontal: 5,
        marginVertical: 10,
        width: '100%',
        borderColor: "#e2e2e2",
        borderWidth: 0.9,
        height: 50
    },
    icon: {
        position: "absolute",
        top: 13,
        right: 18
    }
});

export { PickerAE as Picker };
