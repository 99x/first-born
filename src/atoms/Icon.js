import { Platform } from "react-native";
import React, { Component } from "react";
import Ionicon from "react-native-vector-icons/Ionicons";
import PropTypes from "prop-types";
import { commonColors } from "../utils/color";

export class Icon extends Component {
    render() {
        const { name, size, color, ...otherProps } = this.props;

        if (!name.includes("logo")) {
            return (
                <Ionicon
                    name={
                        Platform.OS === "android" ? "md-" + name : "ios-" + name
                    }
                    size={size}
                    color={color}
                    {...otherProps}
                />
            );
        }
        return (
            <Ionicon name={name} size={size} color={color} {...otherProps} />
        );
    }
}

Icon.propTypes = {
    ...Ionicon.propTypes,
    name: PropTypes.string
};

Icon.defaultProps = {
    color: commonColors.white,
    size: 18
};
