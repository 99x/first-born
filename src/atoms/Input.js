import React, { Component } from "react";
import { TextInput, StyleSheet, Platform, Keyboard, View } from "react-native";
import PropTypes from "prop-types";

import { commonColors } from "../utils/color";

export class Input extends Component {

    constructor(props) {
        super(props);
        this.state = {
            focused: false,
            error: false
        }
    }

    onChangeText = (text) => {
        const { isValid, onChangeText } = this.props;
        onChangeText(text);
        if (isValid) {
            this.setState({ error: isValid(text) });
        }
    }

    render() {
        const { onChangeText, placeholder, color, ...otherProps } = this.props;

        const underLineColor = !color ? commonColors.black : color

        if (Platform.OS === "android") {
            return (
                <View style={this.state.focused ? [styles.input, { borderColor: underLineColor, borderWidth: 2 }] : this.state.error ? [styles.input, { borderColor: "#e74c3c", borderWidth: 2 }] : styles.input}>
                    <TextInput
                        ref={(c) => this._root = c}
                        style={{ width: "100%" }}
                        underlineColorAndroid={"transparent"}
                        placeholder={placeholder}
                        placeholderTextColor={this.state.error ? '#e74c3c' : 'rgba(33, 33, 33, 0.5)'}
                        onChangeText={this.onChangeText}
                        onFocus={() => this.setState({ focused: true })}
                        onBlur={() => this.setState({ focused: false })}
                        onSubmitEditing={() => { this.setState({ focused: true }); Keyboard.dismiss() }}
                        {...otherProps}
                    />
                </View>
            )
        }
        return (
            <TextInput
                ref={(c) => this._root = c}
                style={styles.input}
                underlineColorAndroid={"transparent"}
                placeholder={placeholder}
                onChangeText={onChangeText}
                {...otherProps}
            />
        )
    }
}

Input.propTypes = {
    color: PropTypes.string,
    ...TextInput.propTypes
}

const styles = StyleSheet.create({
    input: {
        borderRadius: 10,
        paddingHorizontal: 5,
        marginVertical: 10,
        width: '100%',
        borderColor: "rgba(33, 33, 33, 0.5)",
        borderWidth: 0.9,
        height: 45
    }
})
