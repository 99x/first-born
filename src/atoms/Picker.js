import React, { Component } from "react";
import { StyleSheet, Platform, Picker, View, Modal } from "react-native";
import createReactClass from "create-react-class";
import Ionicon from "react-native-vector-icons/Ionicons";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

class PickerAE extends Component {

    constructor(props) {
        super(props);
        this.state = {
            focused: false,
            chosenValue: !props.placeholder && props.selectedValue ? props.selectedValue : undefined
        }
    }

    setValue = (value) => {
        this.setState({ chosenValue: value });
        if (this.props.onValueChange) {
            this.props.onValueChange(value);
        }
    }

    render() {
        const { color, placeholder, animationType, modalTransparent, selectedValue, children, ...otherProps } = this.props;
        const { focused, chosenValue } = this.state;

        if (Platform.OS === "android") {
            return (
                <View style={focused ? [styles.pickerAndroid, { borderColor: color, borderWidth: 2 }] : styles.pickerAndroid} >
                    <Picker
                        style={{ height: 45 }}
                        ref={c => (this._root = c)}
                        {...otherProps}
                        selectedValue={chosenValue}
                        onValueChange={this.setValue.bind(this)}
                    >
                        {children}
                    </Picker>
                </View>
            )
        }
        return (
            <View>
                <View style={focused ? [styles.pickerIos, { borderColor: color, borderWidth: 2 }] : styles.pickerIos}>
                    <Text
                        ref={(c) => this._root = c}
                        style={styles.input}
                        onPress={() => this.setState({ focused: true })}
                    >
                        {chosenValue ? chosenValue : placeholder}
                    </Text>
                    <Ionicon onPress={() => this.setState({ focused: true })} name={focused ? "ios-arrow-up" : "ios-arrow-down"} style={styles.icon} color={commonColors.inputGrey} size={24} />
                </View>
                <Modal
                    supportedOrientations={['portrait', 'landscape']}
                    animationType={animationType}
                    transparent={modalTransparent}
                    visible={focused}
                    onRequestClose={() => { }}
                >
                    <Text
                        onPress={() => this.setState({ focused: false })}
                        style={{ flex: 1 }}
                    />
                    <Picker
                        ref={c => (this._root = c)}
                        {...otherProps}
                        selectedValue={chosenValue}
                        onValueChange={this.setValue.bind(this)}
                        style={{ flex: 1, backgroundColor: commonColors.white }}
                    >
                        {children}
                    </Picker>
                    <Text
                        onPress={() => this.setState({ focused: false })}
                        style={{ flex: 1 }}
                    />
                </Modal>
            </View>
        )
    }
}

PickerAE.Item = createReactClass({
    render() {
        return <Picker.Item {...this.props} />;
    }
});

PickerAE.propTypes = {
    ...Picker.propTypes
}

PickerAE.defaultProps = {
    modalTransparent: true,
    color: commonColors.primary,
    animationType: "fade",
    mode: "dropdown",
    placeholder: "Select Option"
}

const styles = StyleSheet.create({
    pickerAndroid: {
        width: "100%",
        marginVertical: 10,
        borderRadius: 10,
        paddingHorizontal: 5,
        borderColor: commonColors.inputGrey,
        borderWidth: 0.9
    },
    pickerIos: {
        width: "100%",
        flexDirection: 'row',
        borderRadius: 15,
        marginVertical: 10,
        borderColor: commonColors.inputGrey,
        borderWidth: 0.9,
        justifyContent: 'center',
        alignItems: 'center',
        height: 45,
        paddingHorizontal: 5
    },
    icon: {
        position: "absolute",
        top: 13,
        right: 18
    },
    input: {
        flex: 1,
        paddingLeft: 10
    }
});

export { PickerAE as Picker };
