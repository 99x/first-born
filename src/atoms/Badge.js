import React, { Component } from "react";
import { View, StyleSheet, Platform, Image } from "react-native";
import PropTypes from "prop-types";
import { Icon } from "./Icon";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

export class Badge extends Component {
    render() {
        const {
            color,
            children,
            ...otherProps
        } = this.props;

        const badgeColor = color;

        let badgeStyle = [styles.defaultBadge, {
            borderRadius: 15,
            backgroundColor: badgeColor
        }];

        let textStyle = {
            color: commonColors.white
        };

        let iconColor = commonColors.white;;

        const newChildren = React.Children.map(children, child =>
            child && child.type === Text
                ? React.cloneElement(child, {
                    ...child.props,
                    style: { ...child.props.style, ...textStyle },
                    size: "footnote"
                })
                : child && child.type === Icon
                    ? React.cloneElement(child, {
                        ...child.props,
                        color: iconColor,
                        size: 15
                    })
                    : child && child.type === Image
                        ? React.cloneElement(child, {
                            ...child.props,
                            style: {
                                ...child.props.style,
                                width: 15,
                                height: 15
                            }
                        })
                        : null
        );

        return (
            <View
                ref={c => (this._root = c)}
                style={badgeStyle}
                {...otherProps}
            >
                {newChildren}
            </View>
        );
    }
}

Badge.propTypes = {
    color: PropTypes.string,
    ...View.propTypes
};

Badge.defaultProps = {
    color: commonColors.primary,
};

const styles = StyleSheet.create({
    defaultBadge: {
        margin: 5,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        paddingHorizontal: 10,
        paddingVertical: 5
    },
});
