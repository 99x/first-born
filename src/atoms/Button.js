import React, { Component } from "react";
import { TouchableOpacity, StyleSheet, Platform, Image } from "react-native";
import PropTypes from "prop-types";
import { Icon } from "./Icon";
import { getFontSize, getButtonPadding, getRoundRadius, getIconSize } from "../variables/buttonSizeVariables";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

export class Button extends Component {
    render() {
        const { title, color, rounded, outline, block, size, secondary, transparent, ...otherProps } = this.props;

        const buttonColor = secondary ? commonColors.secondary : color;

        let buttonStyle = [styles.defaultButton];
        let iconSize = getIconSize(size);
        let textStyle;
        let iconColor;

        if (block) {
            buttonStyle.push(styles.blockButton);
        }

        if (rounded) {
            buttonStyle.push({ paddingHorizontal: getButtonPadding(size) + (getButtonPadding(size) / 3), borderRadius: getRoundRadius(size) });
        }

        if (outline) {
            textStyle = { color: buttonColor, fontSize: getFontSize(size) };
            buttonStyle.push({ borderColor: buttonColor, borderWidth: 1, padding: getButtonPadding(size) });
            iconColor = buttonColor;
        } else {
            textStyle = { color: commonColors.white, fontSize: getFontSize(size) };
            buttonStyle.push({ backgroundColor: buttonColor, padding: getButtonPadding(size) });
            iconColor = commonColors.white;
        }

        if (transparent) {
            buttonStyle.push(styles.transparent);
        }

        const children = React.Children.map(this.props.children, child => child && child.type === Text ?
            React.cloneElement(child, { style: textStyle, ...child.props }) : child && child.type === Icon ?
                React.cloneElement(child, { color: iconColor, size: iconSize, ...child.props }) : child && child.type === Image ?
                    React.cloneElement(child, { style: { width: 25, height: 25 }, ...child.props }) : null);

        return (
            <TouchableOpacity ref={(c) => this._root = c} style={buttonStyle} {...otherProps}>
                {children}
            </TouchableOpacity>
        )
    }
}

Button.propTypes = {
    rounded: PropTypes.bool,
    outline: PropTypes.bool,
    block: PropTypes.bool,
    title: PropTypes.string,
    color: PropTypes.string,
    transparent: PropTypes.bool,
    size: PropTypes.oneOf(["small", "default", "large"]),
    ...TouchableOpacity.propTypes
}

Button.defaultProps = {
    color: commonColors.primary,
    size: "default"
}

const styles = StyleSheet.create({
    text: {
        color: commonColors.white
    },
    defaultButton: {
        ...Platform.select({
            android: {
                borderRadius: 5,
                elevation: 3,
            },
            ios: {
                borderRadius: 10,
                shadowOffset: { width: 3, height: 3 },
                shadowOpacity: 0.3,
                shadowRadius: 3,
            }
        }),
        margin: 10,
        shadowColor: "#000",
    },
    blockButton: {
        width: "100%",
        justifyContent: "center",
        alignItems: "center",
    },
    transparent: {
        borderWidth: 0,
        backgroundColor: "transparent",
        borderRadius: 0,
        shadowOffset: { width: 0, height: 0 },
        shadowOpacity: 0,
        shadowRadius: 0,
        elevation: 0,
    }
})
