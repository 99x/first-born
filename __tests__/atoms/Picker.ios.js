import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Picker } from "../../src/atoms/Picker";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "ios";
    return Platform;
});

it("renders regular Picker", () => {
    const tree = renderer
        .create(
            <Picker>
                <Picker.Item value="1" label="1" />
                <Picker.Item value="2" label="2" />
                <Picker.Item value="3" label="3" />
            </Picker>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders colored Picker", () => {
    const tree = renderer
        .create(
            <Picker color="red">
                <Picker.Item value="1" label="1" />
                <Picker.Item value="2" label="2" />
                <Picker.Item value="3" label="3" />
            </Picker>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders Picker with placeholder", () => {
    const tree = renderer
        .create(
            <Picker placeholder="placeholder">
                <Picker.Item value="1" label="1" />
                <Picker.Item value="2" label="2" />
                <Picker.Item value="3" label="3" />
            </Picker>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
