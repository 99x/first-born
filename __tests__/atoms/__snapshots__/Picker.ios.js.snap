// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Picker with placeholder 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderColor": "rgba(33, 33, 33, 0.5)",
          "borderWidth": 0.9,
        },
        Object {
          "borderRadius": 15,
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      placeholder
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "fontSize": 24,
          },
          Object {
            "paddingRight": 15,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "1",
              "textColor": undefined,
              "value": "1",
            },
            Object {
              "label": "2",
              "textColor": undefined,
              "value": "2",
            },
            Object {
              "label": "3",
              "textColor": undefined,
              "value": "3",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders colored Picker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderColor": "rgba(33, 33, 33, 0.5)",
          "borderWidth": 0.9,
        },
        Object {
          "borderRadius": 15,
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Option
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "fontSize": 24,
          },
          Object {
            "paddingRight": 15,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "1",
              "textColor": undefined,
              "value": "1",
            },
            Object {
              "label": "2",
              "textColor": undefined,
              "value": "2",
            },
            Object {
              "label": "3",
              "textColor": undefined,
              "value": "3",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders custom styled Picker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderColor": "black",
          "borderRadius": 3,
          "borderWidth": 1,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Option
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "fontSize": 24,
          },
          Object {
            "paddingRight": 15,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "1",
              "textColor": undefined,
              "value": "1",
            },
            Object {
              "label": "2",
              "textColor": undefined,
              "value": "2",
            },
            Object {
              "label": "3",
              "textColor": undefined,
              "value": "3",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders regular Picker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderColor": "rgba(33, 33, 33, 0.5)",
          "borderWidth": 0.9,
        },
        Object {
          "borderRadius": 15,
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Option
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "fontSize": 24,
          },
          Object {
            "paddingRight": 15,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "1",
              "textColor": undefined,
              "value": "1",
            },
            Object {
              "label": "2",
              "textColor": undefined,
              "value": "2",
            },
            Object {
              "label": "3",
              "textColor": undefined,
              "value": "3",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders rounded Picker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderColor": "rgba(33, 33, 33, 0.5)",
          "borderWidth": 0.9,
        },
        Object {
          "borderRadius": 15,
          "paddingHorizontal": 10,
        },
        Object {
          "borderRadius": 22.5,
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Option
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "fontSize": 24,
          },
          Object {
            "paddingRight": 15,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "1",
              "textColor": undefined,
              "value": "1",
            },
            Object {
              "label": "2",
              "textColor": undefined,
              "value": "2",
            },
            Object {
              "label": "3",
              "textColor": undefined,
              "value": "3",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders underlined Picker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        Object {
          "borderBottomColor": "rgba(33, 33, 33, 0.5)",
          "borderBottomWidth": 0.9,
        },
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Option
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "fontSize": 24,
          },
          Object {
            "paddingRight": 15,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "1",
              "textColor": undefined,
              "value": "1",
            },
            Object {
              "label": "2",
              "textColor": undefined,
              "value": "2",
            },
            Object {
              "label": "3",
              "textColor": undefined,
              "value": "3",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;

exports[`renders unstyled Picker 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "flex": 1,
          },
        ]
      }
    >
      Select Option
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "rgba(33, 33, 33, 0.5)",
            "fontSize": 24,
          },
          Object {
            "paddingRight": 15,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "1",
              "textColor": undefined,
              "value": "1",
            },
            Object {
              "label": "2",
              "textColor": undefined,
              "value": "2",
            },
            Object {
              "label": "3",
              "textColor": undefined,
              "value": "3",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    />
  </Modal>
</View>
`;
